includes/SpecialManageWikiExtensions.php

```diff
@@ -0,0 +1,213 @@
+<?php
+class SpecialManageWikiExtensions extends SpecialPage {
+	function __construct() {
+		parent::__construct( 'ManageWikiExtensions', 'managewiki-extensions' );
+	}
+
+	function execute( $par ) {
+		global $wgEnableManageWiki, $wgCreateWikiDatabase, $wgDBname;
+
+		$out = $this->getOutput();
+		$this->setHeaders();
+
+		if ( !$wgEnableManageWiki ) {
+			$out->addWikiMsg( 'managewiki-disabled' );
+			return false;
+		}
+
+		$this->checkPermissions();
+
+		if ( $wgCreateWikiDatabase !== $wgDBname ) {
+			$this->showWikiForm( $wgDBname );
+		} elseif ( !is_null( $par ) && $par !== '' ) {
+			$this->showWikiForm( $par );
+		} else {
+			$this->showInputBox();
+		}
+	}
+
+	function showInputBox() {
+		$formDescriptor = array(
+			'dbname' => array(
+				'label-message' => 'managewiki-label-dbname',
+				'type' => 'text',
+				'size' => 20,
+				'required' => true,
+				'name' => 'mwDBname',
+			)
+		);
+
+		$htmlForm = HTMLForm::factory( 'ooui', $formDescriptor, $this->getContext(), 'searchForm' );
+		$htmlForm->setMethod( 'post' )
+			->setSubmitCallback( array( $this, 'onSubmitRedirectToWikiForm' ) )
+			->prepareForm()
+			->show();
+
+		return true;
+	}
+
+	function onSubmitRedirectToWikiForm( array $params ) {
+		global $wgRequest;
+
+		if ( $params['dbname'] !== '' ) {
+			header( 'Location: ' . SpecialPage::getTitleFor( 'ManageWiki' )->getFullUrl() . '/' . $params['dbname'] );
+		} else {
+			return 'Invalid url.';
+		}
+
+		return true;
+	}
+
+	function showWikiForm( $wiki ) {
+		global $wgCreateWikiCategories, $wgCreateWikiUseCategories, $wgManageWikiExtensions, $wgUser, $wgCreateWikiUsePrivateWikis, $wgCreateWikiUseClosedWikis, $wgCreateWikiUseInactiveWikis;
+
+		$out = $this->getOutput();
+
+		$dbName = $wiki;
+
+		$wiki = RemoteWiki::newFromName( $wiki );
+
+		if ( $wiki == NULL ) {
+			$out->addHTML( '<div class="errorbox">' . wfMessage( 'managewiki-missing' )->escaped() . '</div>' );
+			return false;
+		}
+
+		if ( !$this->getRequest()->wasPosted() ) {
+			$out->addWikiMsg( 'managewiki-extensions-header', $dbName );
+		}
+
+		$formDescriptor = array(
+			'dbname' => array(
+				'label-message' => 'managewiki-label-dbname',
+				'type' => 'text',
+				'size' => 20,
+				'default' => $dbName,
+				'disabled' => true,
+				'name' => 'mwDBname',
+			),
+		);
+
+		if ( $wgManageWikiExtensions ) {
```

Comment:
this if should turn into a stanza like L13-16 and be below it otherwise this page is literally useless.

Reply:
So no more if (managewikiextensions) is needed anymore?

Reply:
indeed, though $wgManageWikiExtensions is required to be set so see above comment.